pkg/agent/proxy/proxier.go

```diff
@@ -98,12 +107,6 @@ func (p *proxier) removeStaleServices() {
 			klog.Errorf("Failed to remove flows of Service %v: %v", svcPortName, err)
 			continue
 		}
-		for _, endpoint := range p.endpointsMap[svcPortName] {
```

Comment:
can you explain why we are not calling `removeEndpoint` here?

Reply:
Once a Service was deleted, all its Endpoints will be expired, and removeStaleEndpoints take response for cleaning up, thus we don't need to do removeEndpoint in this function.

I've also added a comment for this function.

Reply:
sounds good, as long as we have good test coverage